### ISSUE TITLE
[Bug] Script not available in docker container

### ISSUE THREAD

**Issue by sandert98** [kyverno/chainsaw#821]
### chainsaw version Version

v0.1.3

### Description

Script is not available when Chainsaw is launched from the container image. 

### Steps to reproduce

1. docker pull ghcr.io/kyverno/chainsaw
2. Mount test files
3. Run chainsaw 

Test step that fails:
```
  steps:
  - try:
    - script: 
        content: |
          kubectl annotate namespace ${NAMESPACE} "egress"="internet"
```

### Expected behavior

Script got executed (in this case, ephemeral namespace should be annotated)

### Logs

```Shell
-c kubectl annotate namespace ${NAMESPACE} "egress"="internet"    
l.go:53:| 09:57:14 | egress-with-annotation | step-1   | SCRIPT    | ERROR |        
=== ERROR        
exec: "sh": executable file not found in $PATH
```

### Slack discussion

https://kubernetes.slack.com/archives/C067LUFL43U/p1707224714140559

### Troubleshooting

- [X] I have searched other issues in this repository and mine is not recorded.

**Comment by eddycharly** [comment]
Thanks for reporting !

**Comment by eddycharly** [comment]
As we are building with `ko` i'm not sure how we can fix this easily 🤔

**Comment by eddycharly** [comment]
@sandert98 this doesn't solve the real problem but in `v0.1.4` you won't need to do that anymore, we added a `namespaceTemplate` to allow modifying the namespace created by chainsaw, see https://github.com/keptn/lifecycle-toolkit/pull/3021

**Comment by sandert98** [comment]
Sounds good indeed, thanks! @eddycharly 
Maybe in the [docs for the script](https://kyverno.github.io/chainsaw/latest/operations/script/) it should be a warning that the script is not available when using the docker container.

**Comment by eddycharly** [comment]
@sandert98 @shubham-cmyk is working on a fix :)

**Comment by eddycharly** [comment]
@sandert98 wanna try with `ghcr.io/kyverno/chainsaw:v0.1.4-beta.1` ?

**Comment by shubham-cmyk** [comment]
@sandert98 it should not be we have verified with e2e test. 
can you share me what you are doing

**Comment by sandert98** [comment]
@shubham-cmyk 

We are trying to test if there is a deprecated API in the cluster before upgrading the k8s cluster. When I run it from my local machine (chainsaw installed with brew), it is working. When I run it within the chainsaw container, it isn't anymore. 

The full test:

```
apiVersion: chainsaw.kyverno.io/v1alpha1
kind: Test
metadata:
  name: deprecated-apis
spec:
  steps:
    - try:
        - script:
            content: |
              kubectl get HorizontalPodAutoscaler --all-namespaces -o=jsonpath='{range .items[*]}{.metadata.namespace}{"\t"}{.metadata.name}{"\t"}{.apiVersion}{"\n"}{end}' | grep v2beta  | wc -l
            check:
              (to_number($stdout) == `0`): true
```

The error can be shown to strip it down a bit:
```
apiVersion: chainsaw.kyverno.io/v1alpha1
kind: Test
metadata:
  name: deprecated-apis
spec:
  steps:
    - try:
        - script:
            content: |
              kubectl get HorizontalPodAutoscaler --all-namespaces -o=jsonpath='{range .items[*]}{.metadata.namespace}{"\t"}{.metadata.name}{"\t"}{.apiVersion}{"\n"}{end}' 
```

**Comment by shubham-cmyk** [comment]
Reopening this I have to test this again. 
i guess the problem was solved. can you make sure that you are using the latest version ?

**Comment by sandert98** [comment]
Thanks!
I have indeed this issue with ghcr.io/kyverno/chainsaw:v0.1.6.

**Comment by eddycharly** [comment]
The approach here using `grep` and `wc` seems wrong to me.

**Comment by eddycharly** [comment]
I would do something like
```yaml
        - script:
            content: |
              kubectl get HorizontalPodAutoscaler --all-namespaces -o=json
            check:
              (json_parse($stdout)):
                # ...
```

**Comment by sandert98** [comment]
When i execute only this part:
```
        - script:
            content: |
              kubectl get HorizontalPodAutoscaler --all-namespaces -o=json
```
I get the error that there is no shell available.

**Comment by eddycharly** [comment]
@sandert98 i reproduced the issue, this might be related to our goreleaser config

**Comment by eddycharly** [comment]
Potential fix https://github.com/kyverno/chainsaw/pull/990

I will cut v0.1.7-alpha.1 tonight to confirm the issue is fixed.

**Comment by eddycharly** [comment]
Released `v0.1.7-alpha.1` and verified, it's fixed. Will be in `v0.1.7` that we're going to release tomorrow.

**Comment by sandert98** [comment]
Thanks @eddycharly ! I can confirm that the issue is indeed been fixed in `v0.1.7-alpha.1`.
